### PR TITLE
Do not produce hdf5 files output when run ecl100

### LINKED ISSUE #1111: ecl100 using eclrun should default to not produce hdf5 files
By default eclrun creates an hdf5 file with data from summary, restart and init files. Currently, nobody in EQNR is using these files and they pile up for no benefit. Pertel is capable of loading the file, but there is limited usage of this. I propose that the ecl100 job defaults to not creating this file by invoking the '--summary-conversion no' option in eclrun.

### PR DESCRIPTION
**Pull request by kvashchuka**
**Issue**
Resolves #1111 

**Approach**
Add flag "--summary-conversion" with default "no" to EclRun

### CODE REVIEW

File: tests/res/fm/test_ecl_run_new.py [hunk 1]
```diff
@@ -163,6 +163,19 @@ def test_failed_run_OK(self):
...
+    def test_no_hdf5_output_with_ecl100(self):
+        self.init_eclrun_config()
+        shutil.copy(
+            os.path.join(self.SOURCE_ROOT, "test-data/local/eclipse/SPE1.DATA"),
+            "SPE1.DATA",
+        )
+        ecl_config = Ecl100Config()
+        run(ecl_config, ["SPE1.DATA", "--version=2019.3"])
+
+        self.assertTrue(os.path.isfile("non-existing-file.HDF5"))
```
Comment (sondreso) [inline tests/res/fm/test_ecl_run_new.py]: BTW, think HDF5 uses small letters in the extension 🙂 
```python
>>> import os
>>> os.path.isfile("setup.py")
True
>>> os.path.isfile("setup.PY")
False
```

Comment (sondreso) [reply tests/res/fm/test_ecl_run_new.py]: And linux uses a case sensitive filesystem by default while mac doesnt

Comment (kvashchuka) [reply tests/res/fm/test_ecl_run_new.py]: I assumed capital letter as in the rest of eclipse output but I might be wrong, will double check, thanks! However, the tests pass anyway and as you can see in the name of the file I want the test to fail..

Comment (sondreso) [reply tests/res/fm/test_ecl_run_new.py]: That is very interesting 🤔

**Comment by markusdregi** [comment]
If we wish to expose the possibility of exporting as hdf5 to our users we should also add the argument in the job description files...

Also, it might be worth thinking about how close to `eclrun` we want to be. If we want to be almost compatible it makes sense to keep it as is, but if not then the more standard use of cli flags would be to have `--summary-conversion` give hdf5, and no flag give no export instead of the `"yes"/"no"` strings...

File: python/res/fm/ecl/ecl_run.py [hunk 2]
```diff
@@ -161,10 +161,13 @@ class EclRun(object):
 
     """
 
-    def __init__(self, ecl_case, sim, num_cpu=1, check_status=True):
+    def __init__(
+        self, ecl_case, sim, num_cpu=1, check_status=True, summary_conversion="no"
```
Comment (markusdregi) [inline python/res/fm/ecl/ecl_run.py]: Either way I suggest we make `summary_conversion` here into a boolean

**Comment by kvashchuka** [comment]
> If we wish to expose the possibility of exporting as hdf5 to our users we should also add the argument in the job description files...
> 
> Also, it might be worth thinking about how close to `eclrun` we want to be. If we want to be almost compatible it makes sense to keep it as is, but if not then the more standard use of cli flags would be to have `--summary-conversion` give hdf5, and no flag give no export instead of the `"yes"/"no"` strings...

That is a very good comment. To be honest, I do not have any insight or preferences on this matter, I just adopted eclrun way of setting the flag

**Comment by markusdregi** [comment]
That is fair. Since, we are reversing default behaviour I suggest we expose it. I just had a closer look and we already expose `OPTS`, which means that users can enable this by passing the flag as part of that argument:

https://github.com/equinor/libres/blob/81bfcb2493b1937fdaa37400b4315322b3ca4586/share/ert/forward-models/old_style/ECLIPSE300#L5

I suggest we move forward with `--summary-conversion` 👍🏻 Also, the job documentation should be updated...

File: share/ert/forward-models/docs/examples/ECLIPSE100.rst [hunk 3]
```diff
@@ -1,9 +1,13 @@
...
 
 .. code-block:: bash
 
-    FORWARD_MODEL ECLIPSE100(<ECLBASE>, <VERSION>=xxxx, <OPTS>="--ignore-errors")
+    FORWARD_MODEL ECLIPSE100(<ECLBASE>, <VERSION>=xxxx, <OPTS>={"--ignore-errors", "--summary-conversion"})
 
-The :code:`OPTS` argument is optional and can be removed, thus running eclipse
-without ignoring errors
+The :code:`OPTS` argument is optional and can be removed, fully or partially.
+In absence of :code:`--ignore-errors` eclipse will run without ignoring errors.
```
Comment (markusdregi) [inline share/ert/forward-models/docs/examples/ECLIPSE100.rst]: `run without ignoring errors` -> `fail on errors`

File: python/res/fm/ecl/ecl_run.py [hunk 4]
```diff
@@ -292,7 +298,6 @@ def execEclipse(self, eclrun_config=None):
                 else self._get_legacy_run_command()
             )
             env = eclrun_config.run_env if use_eclrun else self._get_legacy_run_env()
-
```
Comment (markusdregi) [inline python/res/fm/ecl/ecl_run.py]: Please revert this line change to keep the footprint minimal.

File: tests/res/fm/test_ecl_run_new.py [hunk 5]
```diff
@@ -163,6 +163,23 @@ def test_failed_run_OK(self):
...
+        shutil.copy(
+            os.path.join(self.SOURCE_ROOT, "test-data/local/eclipse/SPE1.DATA"),
+            "SPE1.DATA",
+        )
+        ecl_config = Ecl100Config()
+        # check that by default .h5 file IS NOT produced
+        run(ecl_config, ["SPE1.DATA", "--version=2019.3"])
+        self.assertFalse(os.path.exists("SPE1.h5"))
+
+        # check that with flag .h5 file IS produced
```
Comment (markusdregi) [inline tests/res/fm/test_ecl_run_new.py:188]: Should we split this up into two tests so that the the two states are tested completely independently?

**Review by markusdregi — CHANGES_REQUESTED** [review]
Had a few minor comments. Besides those, this is ready to be merged

File: tests/res/fm/test_ecl_run_new.py (hunk 5 above)
Comment (kvashchuka) [reply tests/res/fm/test_ecl_run_new.py:188]: sure, we can do that

**Review by markusdregi — APPROVED** [review]
Looks good 👍🏻 

Remember to squash your commits...

**Pull request merged**